server/cmd/mmctl/commands/post.go

```diff
@@ -217,3 +230,25 @@ func postListCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	}
 	return multiErr.ErrorOrNil()
 }
+
+func deletePostsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	confirmFlag, _ := cmd.Flags().GetBool("confirm")
+	if !confirmFlag {
+		if err := getConfirmation("Are you sure you want to delete the posts specified?", true); err != nil {
+			return err
+		}
+	}
+	for _, postID := range args {
+		isValidId := model.IsValidId(postID)
+		if !isValidId {
+			printer.PrintError(fmt.Sprintf("Invalid postID: %s", postID))
+			continue
+		}
+		if _, err := c.PermanentDeletePost(context.TODO(), postID); err != nil {
+			printer.PrintError(fmt.Sprintf("Error deleting post: %s. Error: %s", postID, err.Error()))
+			continue
+		}
+		printer.Print(fmt.Sprintf("%s successfully deleted", postID))
+	}
+	return nil
```

Comment:
Could this method return an error so mmctl returns a non-zero error code, as in https://github.com/mattermost/mattermost/pull/26428?